dpnp/linalg/dpnp_algo_linalg.pyx

```diff
@@ -167,50 +167,20 @@ cpdef dparray dpnp_eigvals(dparray input):
     return res_val
 
 
-cpdef dparray dpnp_inv(dparray input):
-    cpdef size_t n = input.shape[0]
-
-    # TODO: replace with dpnp.eye(n) when it will be implemented
-    e_arr = dpnp.diag(dpnp.full((n,), 1, dtype=dpnp.float64))
-    a_arr = input.astype(dpnp.float64)
-
-    for k in range(n):
-        if a_arr[k, k] == 0:
-            for i in range(k, n):
-                if a_arr[i, k] != 0:
-                    for j in range(n):
-                        c = a_arr[k, j]
-                        a_arr[k, j] = a_arr[i, j]
-                        a_arr[i, j] = c
-                        c_e = e_arr[k, j]
-                        e_arr[k, j] = e_arr[i, j]
-                        e_arr[i, j] = c_e
-                    break
-
-        temp = a_arr[k, k]
-
-        for j in range(n):
-            a_arr[k, j] = a_arr[k, j] / temp
-            e_arr[k, j] = e_arr[k, j] / temp
-
-        for i in range(k + 1, n):
-            temp = a_arr[i, k]
-
-            for j in range(n):
-                a_arr[i, j] = a_arr[i, j] - a_arr[k, j] * temp
-                e_arr[i, j] = e_arr[i, j] - e_arr[k, j] * temp
-
-    for k in range(n - 1):
-        ind_k = n - 1 - k
-        for i in range(ind_k):
-            ind_i = ind_k - 1 - i
-
-            temp = a_arr[ind_i, ind_k]
-            for j in range(n):
-                a_arr[ind_i, j] = a_arr[ind_i, j] - a_arr[ind_k, j] * temp
-                e_arr[ind_i, j] = e_arr[ind_i, j] - e_arr[ind_k, j] * temp
-
-    return e_arr
+cpdef dparray dpnp_inv(dparray input_):
+    cdef dparray input = input_.astype(dpnp.float64)
+    cdef DPNPFuncType param1_type = dpnp_dtype_to_DPNPFuncType(input.dtype)
+
+    cdef DPNPFuncData kernel_data = get_dpnp_function_ptr(DPNP_FN_INV, param1_type, param1_type)
+
+    cdef dparray result = dparray(input.size, dtype=dpnp.float64)
```

Comment:
Are you sure you need fixed return type `dpnp.float64`?
Perhaps it is better to use `kernel_data.return_type` as above?
if you kernel must return `float64` it might be better to use `kernel_data.return_type` and put it into `{eft_DBL, (void*)dpnp_inv_c<double>};` for all input types.

Reply:
@shssf I'm sure that this need. Because with this type in result array algorithm for inv in cython start work.